documentation/guides/migrating-from-v5-to-v6.md

```diff
@@ -0,0 +1,251 @@
+# Migrating from v5 to v6
+
+Polaris v6.0.0 ([full release notes](https://github.com/Shopify/polaris-react/releases/tag/v6.0.0)) removes deprecated features in past releases. This file describes all code updates required to stay up to date.
+
+## AppProvider theming changes
+
+### newDesignLanguage feature has been removed and is now the default
+
+In v6 you no longer need to enable the newDesignLanguage through the features flag in the `AppProvider`.
+
+```diff
+<AppProvider
+  i18n={{}}
+- features={{newDesignLanguage: true}}
+>
+```
+
+### Removed legacy topbar theming
+
+With the new visual styles the TopBar theming has been replaced with the `ThemeProvider`. If you want to change the `TopBar` appearance you can ovveride the theme or the CSS variables.
+
+```diff
+<AppProvider
+  i18n={{}}
+  theme={{
+    colors: {
+-     topBar: {
+-       color: 'rgb(33,43,54)',
+-       backgroundLighter: 'rgb(244, 246, 248)',
+-       backgroundDarker: 'rgb(223, 227, 232)',
+-       background: 'rgb(255,255,255)',
+-     },
++     surface: 'rgb(0, 0, 0)',
++     onSurface: 'rgb(33,43,54)',
+    },
+  }}
+>
+```
+
+### Changed `frameOffset` prop to accept a string
+
+This allows relative units for the frame offset. Allowing users to take into account for font-size and matching up breakpoints.
+
+```diff
+<AppProvider
+  i18n={{}}
+  theme={{
+-   frameOffset: 60,
++   frameOffset: '60px',
+  }}
+>
+```
+
+## Exported type changes
+
+### Removed exported type`AnimationProps`
+
+`AnimationProps` is a simple type that was used once in `@shopify/polaris` and we have decided to remove it.
+
+```diff
+- import {AnimationProps} from '@shopify/polaris';
+
++ interface AnimationProps {
++   in?: boolean;
++ }
+```
+
+### Removed export for `NewDesignLanguageColor` and `Color`
+
+We no longer support `NewDesignLanguageColor` and `Color` types. They were used to type our icon colors and have been replaced with ` IconProps['color']`.
```

Comment:
This feels like a kinda odd wording - it's not that people should use IconProps['color'] instead oof Color/NewDesignLanguageColor, it's that they shouldn't need to access these colors in the first place. [heres how a similiar concept was described previously](https://github.com/Shopify/polaris-react/blob/main/documentation/guides/migrating-from-v3-to-v4.md#linklikecomponent-)

```suggestion
The `NewDesignLanguageColor` and `Color` that describes the argument you pass into Icon's color prop. Yoou should use `IconProps['color']` instead of referencing these low level types.
```